### PR TITLE
Add more rest config generation options.

### PR DESCRIPTION
**Pull request by briantkennedy**
* Add function that will try to auto connect
* Add function that will connect from inside a pod

---
This change is [Reviewable](https://reviewable.io/reviews/mdruskin/kubernetes-enterprise-control/41)

### CODE REVIEW

File: pkg/client/restconfig/restconfig.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+var flagRestConfigSource = flag.String(
+	"restConfigSource",
+	"",
+	"Rest config source for the NewRestConfig call, values minikube, podServiceAccount, kubectl")
+
+// NewRestConfig will attempt to create a new rest config from all configured options and return
+// the first successfully created configuration.
+func NewRestConfig() (*rest.Config, error) {
+	switch *flagRestConfigSource {
+	case "minkube":
```
Comment (filmil) [inline pkg/client/restconfig/restconfig.go]: typo: minkube.

File: pkg/client/restconfig/restconfig.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	case "podServiceAccount":
+		return NewPodServiceAccountConfig()
+	case "kubectl":
+		return NewKubectlConfig()
+	case "":
+	default:
+		panic(errors.Errorf("No rest config source named %s", *flagRestConfigSource))
+	}
+
+	for _, configCreator := range []func() (*rest.Config, error){
```
Comment (filmil) [inline pkg/client/restconfig/restconfig.go]: Is this code reachable?  The switch above seems to either return  or panic.  What am I not seeing?

Comment (briantkennedy) [reply pkg/client/restconfig/restconfig.go]: line 40: case "":

File: pkg/client/restconfig/restconfig.go (hunk 1 above)
Comment (briantkennedy) [reply pkg/client/restconfig/restconfig.go]: done

**Comment by filmil** [comment]
:lgtm:

---

Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv48AdmiyqGfKtxbffR:be5trej) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, briantkennedy (Brian Kennedy) wrote…</i></summary><blockquote>

line 40: case "":
</blockquote></details>

OK

Ah yes..., 
but, care to add `// do nothing` here for those of us who forget how golang handles fallthrough?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-:-Kv48bXNZ7fWBvBdtQmS:bnfp4nl)*

**Comment by briantkennedy** [comment]
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv4DssFE4ndX7VkYwxi:b-v63iwp) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, filmil (Filip Filmar) wrote…</i></summary><blockquote>

OK

Ah yes..., 
but, care to add `// do nothing` here for those of us who forget how golang handles fallthrough?
</blockquote></details>

Changed control flow so this is more obvious and added more logging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41)*

**Comment by filmil** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/client/restconfig/restconfig.go, line 45 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-Kv3ru818rSNU9GabE93-r1-45:-Kv4EjAsO1mo-jyI38nt:bsqjb9o) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/f0abc3b986d59be526bbca86936ec47265099cba/pkg/client/restconfig/restconfig.go#L45)):*
<details><summary><i>Previously, briantkennedy (Brian Kennedy) wrote…</i></summary><blockquote>

Changed control flow so this is more obvious and added more logging.
</blockquote></details>

LGTM

Neat!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/41#-:-Kv4EcgUCMbZ-ru0H2id:bnfp4nl)*

**Pull request merged**